### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by redacuve**
This is the pull request of the enumerable methods

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress :+1: @redacuve 
You have made a nice progress with this project given that it's your first PR. I would like to highlight some failing edge cases that we need to fix before we can merge.
See my comments and make those changes before submitting the PR for another review.

### Martin Afani

**Comment by redacuve** [comment]
> ## Nice Progress  @redacuve
> You have made a nice progress with this project given that it's your first PR. I would like to highlight some failing edge cases that we need to fix before we can merge.
> See my comments and make those changes before submitting the PR for another review.
> 
> ### Martin Afani

Where are your comments? thanks

**Comment by redacuve** [comment]
> ## Nice Progress  @redacuve
> You have made a nice progress with this project given that it's your first PR. I would like to highlight some failing edge cases that we need to fix before we can merge.
> See my comments and make those changes before submitting the PR for another review.
> 
> ### Martin Afani

Where are your comments? thanks

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] my_each returns an Enumerator if no block is given
You can do something like this;
`def my_each
   return to_enum unless block_given?
   while i < length
        yield([i])
        i += 1
      end
end`

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    if block_given?
+      i = 0
+      while i < length
+        yield(self[i], i)
+        i += 1
+      end
+    else
+      'No block given'
+    end
+  end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] returns an enumerator if no block is given. Do the same as above :point_up:

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  def my_select
+    ary = []
+    if block_given?
+      my_each { |elem| ary.push(elem) if yield(elem) }
+    else
+      'No block given'
+    end
+    ary
+  end
+
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] returns an enumerator if no block is given

File: enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  def my_all?
+    if block_given?
+      my_each { |elem| return false unless yield(elem) }
+    else
+      my_each { |elem| return false unless elem }
+    end
+    true
+  end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] when a class is passed as an argument returns true if all of the collection is a member of such class
- [x] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [x]  when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

File: enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+  def my_any?
+    if block_given?
+      my_each { |elem| return true if yield(elem) }
+    else
+      my_each { |elem| return true if elem }
+    end
+    false
+  end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [x] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [x] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

File: enumerable_methods.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  def my_none?
+    if block_given?
+      my_each { |elem| return false if yield(elem) }
+    else
+      my_each { |elem| return false if elem }
+    end
+    true
+  end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] when a class is passed as an argument returns true if none of the collection is a member of such class
- [x] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [x] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

File: enumerable_methods.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+  def my_count
+    c = 0
+    if block_given?
+      my_each { |elem| c += 1 if yield(elem) }
+    else
+      my_each { c += 1 }
+    end
+    c
+  end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] counts the number of items in enum that are equal to item if an argument is given

File: enumerable_methods.rb [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  def my_inject(cont = nil)
+    if block_given?
+      cont ||= 0
+      my_each { |elem| cont = yield(cont, elem) }
+      cont
+    else
+      'No block Given (LocalJumpError)'
+    end
+  end
+end
```
Comment (whiz25) [inline enumerable_methods.rb]: - [x] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [x] combines all elements of enum by applying a binary operation, specified by a block
- [x] when a symbol is specified combines each element of the collection by applying the symbol as a named method

**Comment by redacuve** [comment]
image
I got these lint errors and i don't know how to fix that, all is working, I need help with this, please.

**Comment by Anna-Myzukina** [comment]
### Pull Request under the review by Anna Muzykina (TSE)

**Comment by Anna-Myzukina** [comment]
### Your project looks :+1:  good!
### Great job thus far. Please fix Lint errors before it can be approved.

stickler


### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*

**Comment by redacuve** [comment]
> ### Your project looks  good!
> ### Great job thus far. Please fix Lint errors before it can be approved.
> stickler
> 
> ### Make changes and submit another code review request.
> _If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you._

thank you I know, how can I fix these errors???

**Comment by redacuve** [comment]
I need help with that, please!

**Comment by Anna-Myzukina** [comment]
# Status [ APPROVED] image

### You can merge it. And don't forget to fill the project completion form.

Kindly recommended
screen

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*

**Pull request merged**